cmd/snap/cmd_run.go

```diff
@@ -175,9 +177,17 @@ func runSnapConfine(info *snap.Info, securityTag, snapApp, command, hook string,
 		logger.Noticef("WARNING: cannot create user data directory: %s", err)
 	}
 
-	cmd := []string{
-		filepath.Join(dirs.LibExecDir, "snap-confine"),
+	// use snap-confine from core if it is available
+	snapConfinePath := filepath.Join(dirs.LibExecDir, "snap-confine")
+	snapConfinePathInCore := filepath.Join(dirs.SnapMountDir, "/core/current/", snapConfinePath)
+
+	cmd := []string{}
+	if release.OnClassic && osutil.FileExists(snapConfinePathInCore) {
+		cmd = append(cmd, snapConfinePathInCore)
```

Comment:
Unconditionally using snapConfinePathInCore seems problematic. Do we unconditionally use everything else from core there? What about reverts?